src/server.rs

```diff
@@ -438,6 +443,94 @@ impl Server {
         Error::verify_good(&status_code)
     }
 
+    /// Add a method node to the address space.
+    ///
+    /// does something
+    ///
+    /// do not supply type definition
+    ///
+    /// missing parameters:
+    /// UA_MethodCallback method,
+    /// size_t inputArgumentsSize, const UA_Argument *inputArguments,
+    /// const UA_NodeId inputArgumentsRequestedNewNodeId,
+    /// UA_NodeId *inputArgumentsOutNewNodeId,
+    /// size_t outputArgumentsSize, const UA_Argument *outputArguments,
+    /// const UA_NodeId outputArgumentsRequestedNewNodeId,
+    /// UA_NodeId *outputArgumentsOutNewNodeId
+    ///
+    pub fn add_method_node(
+        &self,
+        node: MethodNode,
+        callback: impl MethodCallback + 'static,
+    ) -> Result<(ua::NodeId, Option<MethodNodeArgumentsNodeIds>)> {
+        let MethodNode {
+            requested_new_node_id,
+            parent_node_id,
+            reference_type_id,
+            browse_name,
+            attributes,
+            input_arguments,
+            output_arguments,
+            arguments_request_new_node_ids,
+        } = node;
+
+        let (raw_callback, context) = unsafe { wrap_callback(callback) };
+        let (input_arguments_size, input_arguments) = unsafe { input_arguments.as_raw_parts() };
+        let (output_arguments_size, output_arguments) = unsafe { output_arguments.as_raw_parts() };
+
+        let (arguments_request_new_node_ids, is_extended_call) =
+            if arguments_request_new_node_ids.is_none() {
+                (
+                    MethodNodeArgumentsNodeIds {
+                        input: ua::NodeId::null(),
+                        output: ua::NodeId::null(),
+                    },
+                    false,
+                )
+            } else {
+                (arguments_request_new_node_ids.expect(""), true)
+            };
+
+        let mut arguments_out_node_ids = MethodNodeArgumentsNodeIds::init();
+
+        // This out variable must be initialized without memory allocation because the call below
+        // overwrites it in place, without releasing any held data first.
+        let mut out_node_id = ua::NodeId::null();
+        let status_code = ua::StatusCode::new(unsafe {
+            UA_Server_addMethodNodeEx(
```

Comment:
My approach is to combine addMethodNode and addMethodNodeEx. 
It requires a optional field in the struct and some handling inside the add_method_node function, but I think this is better than having two separate functions.

Reply:
What do you think of that approach @sgoll ?